/workflow crashes on windows

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

open zed on windows
open the ai chat
try using /workflow command

### Environment

Zed: v0.151.0 (Zed Dev 76476446029cfb20558eaaa0b3477113174e604b)
OS: Windows 10.0.22631
Memory: 47.9 GiB
Architecture: x86_64
GPU: NVIDIA GeForce GTX 1070 Ti || NVIDIA || 560.70

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-08-24T18:44:10.2366355+02:00 [ERROR] Error { code: HRESULT(0x80070490), message: "Element not found." }
<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/assistant/src/prompts.rs
+++ b/crates/assistant/src/prompts.rs
@@ -8,6 +8,7 @@ use language::BufferSnapshot;
 use parking_lot::Mutex;
 use serde::Serialize;
 use std::{ops::Range, path::PathBuf, sync::Arc, time::Duration};
+use text::LineEnding;
 use util::ResultExt;
 
 #[derive(Serialize)]
@@ -190,9 +191,10 @@ impl PromptBuilder {
         for path in Assets.list("prompts")? {
             if let Some(id) = path.split('/').last().and_then(|s| s.strip_suffix(".hbs")) {
                 if let Some(prompt) = Assets.load(path.as_ref()).log_err().flatten() {
+                    let mut prompt = String::from_utf8_lossy(prompt.as_ref()).to_string();
+                    LineEnding::normalize(&mut prompt);
                     log::info!("Registering built-in prompt template: {}", id);
-                    handlebars
-                        .register_template_string(id, String::from_utf8_lossy(prompt.as_ref()))?
+                    handlebars.register_template_string(id, prompt)?
                 }
             }
         }

--- a/crates/assistant/src/prompts.rs
+++ b/crates/assistant/src/prompts.rs
@@ -191,10 +191,9 @@ impl PromptBuilder {
         for path in Assets.list("prompts")? {
             if let Some(id) = path.split('/').last().and_then(|s| s.strip_suffix(".hbs")) {
                 if let Some(prompt) = Assets.load(path.as_ref()).log_err().flatten() {
-                    let mut prompt = String::from_utf8_lossy(prompt.as_ref()).to_string();
-                    LineEnding::normalize(&mut prompt);
                     log::info!("Registering built-in prompt template: {}", id);
-                    handlebars.register_template_string(id, prompt)?
+                    let prompt = String::from_utf8_lossy(prompt.as_ref());
+                    handlebars.register_template_string(id, LineEnding::normalize_cow(prompt))?
                 }
             }
         }

--- a/crates/text/src/text.rs
+++ b/crates/text/src/text.rs
@@ -3041,4 +3041,12 @@ impl LineEnding {
             text
         }
     }
+
+    pub fn normalize_cow(text: Cow<str>) -> Cow<str> {
+        if let Cow::Owned(replaced) = LINE_SEPARATORS_REGEX.replace_all(&text, "\n") {
+            replaced.into()
+        } else {
+            text
+        }
+    }
 }

